pipelines/batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -285,7 +300,13 @@ static Pipeline buildHapiJdbcFetch(
       if (numResources == 0) {
         continue;
       }
+
       foundResource = true;
+      PCollection<KV<String, Long>> initialPCollection =
+          pipeline.apply(
+              "Metric parameters for " + resourceType,
+              Create.of(KV.of(resourceType, Long.valueOf(numResources))));
+      initialPCollection.apply(ParDo.of((new LogAsMetric())));
```

Comment:
This is an interesting way of injecting these values into the `MetricResults` of the pipeline but is there no easier way of achieving this without a `DoFn`? I mean these form a small set of resourceType->number maps which can be passed to any `DoFn`, e.g., as constructor parameter, would that option make this easier to achieve?

Reply:
If you can address it before merge it would be nice. I don't mean to actually do what I suggested but analyze that approach and make a decision whether to go with this or change it to that (i.e., remove the TODO in either case).

Reply:
The metrics could not be logged when injected via the constructor of the beginning pipeline stage i.e. in the JdbcFetchHapi.generateQueryParameters() method (looks like the pipeline context is not available at this stage). Hence retaining the current implementation.